content/kb/troubleshooting-instance-manager-pods-are-restarted.md

```diff
@@ -0,0 +1,75 @@
+---
+title: "Troubleshooting: Instance manager pods are restarted"
+authors:
+- "Jack Lin"
+draft: false
+date: 2024-12-12
+versions:
+- "all"
+categories:
+- "instance manager"
+---
+
+## Applicable versions
+
+All Longhorn versions.
+
+## Symptoms
+
+The instance manager pods are restarted which leads to a huge amount of iSCSI connection errors.
+As the consequence, Longhorn engines lose the connection to the replicas and it makes the volume unstable.
+
+You might find iSCSI errors in the Kernel log
+```
+Nov 21 00:54:02 node-xxx kernel:  connection438:0: detected conn error (1020)
+Nov 21 00:54:02 node-xxx kernel:  connection437:0: detected conn error (1020)
+Nov 21 00:54:02 node-xxx kernel:  connection436:0: detected conn error (1020)
+Nov 21 00:54:02 node-xxx kernel:  connection435:0: detected conn error (1020)
+Nov 21 00:54:02 node-xxx kernel:  connection434:0: detected conn error (1020)
+Nov 21 00:54:02 node-xxx kernel:  connection433:0: detected conn error (1020)
+Nov 21 00:54:02 node-xxx kernel:  connection432:0: detected conn error (1020)
+....
+Nov 21 00:54:02 node-xxx kernel:  connection275:0: detected conn error (1020) 
+```
+
+You might also find that the instance manager container suddenly terminates with logs
+```
+time="2024-11-21T06:12:20.651526777Z" level=info msg="shim disconnected" id=548c02c5bc17426da586373f902e8d5811d5efe4e45d5fbd0495920626d014d9 namespace=k8s.io
+time="2024-11-21T06:12:20.651603253Z" level=warning msg="cleaning up after shim disconnected" id=548c02c5bc17426da586373f902e8d5811d5efe4e45d5fbd0495920626d014d9 namespace=k8s.io
+time="2024-11-21T06:12:21.819863412Z" level=info msg="Container to stop \"548c02c5bc17426da586373f902e8d5811d5efe4e45d5fbd0495920626d014d9\" must be in running or unknown state, current state \"CONTAINER_EXITED\""
+```
+
+## Reason
+
+The instance manager pod is a critical component responsible for managing the engine and replica processes of the volume. If the instance manager pod becomes unstable and crashes or restarts unexpectedly, the volumes will also become unstable.
+
+Here are some common reasons reasons why an instance manager pod is restarted or deleted.
+
+### High CPU Loading
+
+The instance manager pod has a liveness probe configured to periodically check the health of the servers inside the pod.
+One potential cause of instability is an excessive number of replica and engine processes running within the pod, which can overload the server and prevent it from responding to the liveness probe in time.
```

Comment:
```suggestion
One potential cause of instability is an excessive number of replica or engine processes running within the pod. This can overload the server and prevent it from responding to the liveness probe in time.
```

Reply:
fixed.